### PR TITLE
[chore] add linux package tests

### PR DESCRIPTION
**Pull request by mowies**
### This PR
- adds tests on amd64 for the linux packages (deb and rpm) using container images that are run with podman
- adds a readme for running the tests locally
- adds the tests to the already existing snapshot pipeline

### CODE REVIEW

File: internal/testbed/linux-services/package-tests.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# ensure service has started and still running after 5 seconds
+sleep 5
+echo "Checking $SERVICE_NAME service status ..."
+$container_exec systemctl --no-pager status "$SERVICE_NAME"
+
+echo "Checking $PROCESS_NAME process ..."
+$container_exec pgrep -f -a -u otel "$PROCESS_NAME"
+
+# test uninstall
+echo
```
Comment (odubajDT) [inline internal/testbed/linux-services/package-tests.sh:60]: I guess something should be printed out here, right?

Comment (mowies) [reply internal/testbed/linux-services/package-tests.sh:60]: this came from upstream, i think we should just remove it

Comment (mowies) [reply internal/testbed/linux-services/package-tests.sh:60]:
```suggestion
```

**Review by odubajDT — APPROVED** [review]

**Review by bacherfl — APPROVED** [review]

File: internal/testbed/linux-services/package-tests.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+image_name="otelcontribcol-$pkg_type-test"
+container_name="$image_name"
+container_exec="podman exec $container_name"
+
+trap 'podman rm -fv $container_name >/dev/null 2>&1 || true' EXIT
+
+podman build -t "$image_name" -f "$SCRIPT_DIR/Dockerfile.test.$pkg_type" "$SCRIPT_DIR"
+podman rm -fv "$container_name" >/dev/null 2>&1 || true
+
+podman run --name "$container_name" -d "$image_name"
```
[the remark this answers is not in the document]
Comment (mowies) [reply internal/testbed/linux-services/package-tests.sh:47]: no specific reason, this is how it's done upstream

**Review by heckelmann — APPROVED** [review]
lgtm

**Pull request merged**